cpp/Platform.Delegates/Platform.Delegates.MulticastDelegate.h

```diff
@@ -126,57 +88,98 @@ namespace Platform::Delegates
 
         MulticastDelegate &operator+=(const DelegateType &callback)
         {
-            const std::lock_guard<std::mutex> lock(mutex);
+            std::lock_guard lock(mutex);
             this->callbacks.emplace_back(callback);
             return *this;
         }
 
-        MulticastDelegate &operator+=(DelegateRawFunctionType &callback)
+        MulticastDelegate &operator+=(DelegateRawFunctionType callback)
         {
-            return *this += DelegateType(callback);
+            return *this += DelegateType{callback};
         }
 
         MulticastDelegate &operator+=(const DelegateFunctionType &callback)
         {
-            return *this += DelegateType(callback);
+            return *this += DelegateType{callback};
         }
 
         MulticastDelegate &operator-=(const DelegateType &callback)
         {
-            const std::lock_guard<std::mutex> lock(mutex);
+            const std::lock_guard lock{mutex};
             auto searchResult = std::find(this->callbacks.rbegin(), this->callbacks.rend(), callback);
             if (searchResult != this->callbacks.rend()) {
-                auto removedCallbackPosition = --(searchResult.base());
-                this->callbacks.erase(removedCallbackPosition);
+                this->callbacks.erase(searchResult.base());
             }
             return *this;
         }
 
         MulticastDelegate &operator-=(DelegateRawFunctionType &callback)
         {
-            return *this -= DelegateType(callback);
+            return *this -= DelegateType{callback};
         }
 
         MulticastDelegate &operator-=(const DelegateFunctionType &callback)
         {
-            return *this -= DelegateType(callback);
+            return *this -= DelegateType{callback};
```

Comment:
What is the difference in this context? Why curly brackets are better?

Reply:
This is the preferred way of initializing objects in C ++, you can see the details here -> https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax

Reply:
Ok, let's try it.